dask/order.py

```diff
@@ -279,6 +290,9 @@ def finish_now_key(x):
                 item = inner_stack_pop()
                 if item in result:
                     continue
+                if skip_root_node and item in root_nodes:
```

Comment:
I'm curious, how will this item get processed if we always skip it?  Actually, I think I know the answer: it will get processed when we handle `finish_now` down below.

Reply:
I guess the answer is *sometimes*. As you may see I had to cancel the CI since we got caught in some infinite loops. Hopefully I've caught all those through trial and error.

Reply:
Interesting.  In that case, this check could probably go a couple lines lower inside the `if num_needed[item]:` branch.